### PR TITLE
feat(WAF): waf cloud support postpaid

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:

*** waf cloud support postpaid

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

- Add a new parameter `website`.
- Change **Required** parameters to **Optional**: `resource_spec_code`, `period_unit`, `period`.
- Make the resource support both `prePaid` and `postPaid`.
- Change test, add two unit test for `postPaid`, add one unit test for validation check.

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccCloudInstance_prepaid_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccCloudInstance_prepaid_basic -timeout 360m -parallel 4 
=== RUN   TestAccCloudInstance_prepaid_basic 
=== PAUSE TestAccCloudInstance_prepaid_basic
=== CONT  TestAccCloudInstance_prepaid_basic
--- PASS: TestAccCloudInstance_prepaid_basic (132.44s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       132.506s
```

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccCloudInstance_prepaid_withEpsID'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccCloudInstance_prepaid_withEpsID -timeout 360m -parallel 4 
=== RUN   TestAccCloudInstance_prepaid_withEpsID 
=== PAUSE TestAccCloudInstance_prepaid_withEpsID
=== CONT  TestAccCloudInstance_prepaid_withEpsID
--- PASS: TestAccCloudInstance_prepaid_withEpsID (145.19s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       145.265s
```
```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccCloudInstance_postpaid_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccCloudInstance_postpaid_basic -timeout 360m -parallel 4 
=== RUN   TestAccCloudInstance_postpaid_basic 
=== PAUSE TestAccCloudInstance_postpaid_basic
=== CONT  TestAccCloudInstance_postpaid_basic
--- PASS: TestAccCloudInstance_postpaid_basic (10.80s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       10.871s
```
```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccCloudInstance_postpaid_withEpsID'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccCloudInstance_postpaid_withEpsID -timeout 360m -parallel 4 
=== RUN   TestAccCloudInstance_postpaid_withEpsID 
=== PAUSE TestAccCloudInstance_postpaid_withEpsID
=== CONT  TestAccCloudInstance_postpaid_withEpsID
--- PASS: TestAccCloudInstance_postpaid_withEpsID (10.48s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       10.548s
```
```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccCloudInstance_validation'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccCloudInstance_validation -timeout 360m -parallel 4 
=== RUN   TestAccCloudInstance_validation 
=== PAUSE TestAccCloudInstance_validation
=== CONT  TestAccCloudInstance_validation
--- PASS: TestAccCloudInstance_validation (5.21s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       5.278s
```

### CODE REVIEW

File: huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go [hunk 1]
```diff
@@ -181,6 +176,84 @@ func resourceCloudInstanceCreate(ctx context.Context, d *schema.ResourceData, me
...
+}
+
+func ValidatePrePaidParameter(d *schema.ResourceData) error {
+	if _, ok := d.GetOk("website"); ok {
+		return fmt.Errorf("`website` cannot be specified in pre paid charging mode")
+	}
+	return nil
+}
+
+func ValidatePostPaidParameter(d *schema.ResourceData) error {
```
Comment (Lance52259) [inline huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go]: These functions does not used outside of this package, please change these function protection level to the `private`.

Comment (deer-hang) [reply huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go]: Done

File: docs/resources/waf_cloud_instance.md [hunk 2]
```diff
@@ -32,50 +34,90 @@ resource "huaweicloud_waf_cloud_instance" "test" {
...
-* `resource_spec_code` - (Required, String) Specifies the specification of the cloud WAF.  
+* `charging_mode` - (Required, String, ForceNew) Specifies the charging mode of the cloud WAF.
+  The valid values are **postPaid** and **prePaid** (the yearly/monthly billing mode).
+
+  + If `charging_mode` is set to **postPaid**, only `region`, `website`, `enterprise_project_id` can be specified.
+  + If `charging_mode` is set to **prePaid**, `website` cannot be specified.
+
+  Changing this will create a new resource.
+
+  -> The postpaid charging mode is only applicable to Huawei Cloud International website.
```
Comment (ShiChangkuo) [inline docs/resources/waf_cloud_instance.md]: how about move L61 to here?

File: huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go [hunk 3]
```diff
@@ -181,6 +176,84 @@ func resourceCloudInstanceCreate(ctx context.Context, d *schema.ResourceData, me
...
+	}
+	return nil
+}
+
+func validatePostPaidParameter(d *schema.ResourceData) error {
+	if _, ok := d.GetOk("website"); !ok {
+		return fmt.Errorf("`website` must be specified in postpaid charging mode")
+	}
+
+	invalidFields := []string{"resource_spec_code", "period_unit", "period", "auto_renew", "bandwidth_expack_product",
```
Comment (ShiChangkuo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go]: how about using `ConflictsWith` in the definition of website

File: huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go [hunk 4]
```diff
@@ -397,6 +476,11 @@ func updateExtendedPackages(ctx context.Context, wafClient *golangsdk.ServiceCli
 }
 
 func resourceCloudInstanceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if d.Get("charging_mode").(string) == string(ChargingModePostPaid) {
+		return diag.Errorf("the postpaid charging mode cloud WAF do not support update. Please remove the" +
+			" useless parameters.")
```
Comment (ShiChangkuo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go]:
```suggestion
		return diag.Errorf("the postpaid charging mode cloud WAF instances cannot be updated.")
```

File: docs/resources/waf_cloud_instance.md (hunk 2 above)
Comment (deer-hang) [reply docs/resources/waf_cloud_instance.md]: OK

File: huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go (hunk 3 above)
Comment (deer-hang) [reply huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go]: OK

File: huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go (hunk 4 above)
Comment (deer-hang) [reply huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go]: OK

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccCloudInstance_validation'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccCloudInstance_validation -timeout 360m -parallel 4 
=== RUN   TestAccCloudInstance_validation 
=== PAUSE TestAccCloudInstance_validation
=== CONT  TestAccCloudInstance_validation
--- PASS: TestAccCloudInstance_validation (2.56s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       2.626s
```

**Pull request merged**